homework-1/task_2.py

```diff
@@ -0,0 +1,72 @@
+
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code.
+Feel free to add any more new features to your class.
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student.
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id, subject):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = {
+            "name": name,
+            "subject":subject
+        }
+
+class CFGstudent(Student):
+
+    def __init__(self, name, age, id, subject):
+        super().__init__()
+    def add_subject(self, subject):
+        self.subjects = {}
+        self.subjects.update({"subject"}) == subject
+        print(f"A subject {self.subjects[1]} has been added to your timetable.")
+
+    def remove_subject(self, subject):
+        self.subjects = {}
+        self.subjects.pop(subject)
+        print(f"A subject {self.subjects[subject]} has been removed from your timetable.")
```

Comment:
Inside the `print` statement, interpolating `subject` rather than `self.subjects[subject]` would have been more likely to work in this case since you just popped the subject from `self.subjects` on the line before